libnavigation-util/src/test/java/com/mapbox/navigation/utils/timer/MapboxTimerTest.kt

```diff
@@ -1,94 +1,114 @@
-        advanceTimeBy(150)
-
-        assertFalse(lambdaCalled)
+        coVerify(exactly = 5) { mockLambda.invoke() }
     }
 
     @Test
-    fun executeLambda_notCalled_when_startNotCalled() = runBlockingTest {
-        var lambdaCalled = false
-        val testLambda = { lambdaCalled = true }
+    fun `should update interval dynamically`() = coroutineRule.runBlockingTest {
+        mapboxTimer.routeRefreshInterval = TimeUnit.MINUTES.toMillis(1)
+        val job = mapboxTimer.startRouteRefresh(mockLambda)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(1))
+        mapboxTimer.routeRefreshInterval = TimeUnit.MINUTES.toMillis(10)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(10))
+        job.cancelAndJoin()
+
+        coVerify(exactly = 2) { mockLambda.invoke() }
+    }
 
-        MapboxTimer(0L, testLambda)
-        advanceTimeBy(10)
+    @Test
+    fun `should stop when canceled`() = coroutineRule.runBlockingTest {
+        mapboxTimer.routeRefreshInterval = TimeUnit.MINUTES.toMillis(5)
+        val job = mapboxTimer.startRouteRefresh(mockLambda)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(6))
+        mapboxTimer.stopJobs()
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(30))
+        job.cancelAndJoin()
+
+        coVerify(exactly = 1) { mockLambda.invoke() }
+    }
 
-        assertFalse(lambdaCalled)
+    @Test
+    fun `should subscribe to jobs separately`() = coroutineRule.runBlockingTest {
+        val mockLambda1: () -> Unit = mockk(relaxed = true)
+        val mockLambda2: () -> Unit = mockk(relaxed = true)
+
+        mapboxTimer.routeRefreshInterval = 5000
+        val job1 = mapboxTimer.startRouteRefresh(mockLambda1)
+        coroutineRule.testDispatcher.advanceTimeBy(5001)
+        val job2 = mapboxTimer.startRouteRefresh(mockLambda2)
+        coroutineRule.testDispatcher.advanceTimeBy(5001)
+        job1.cancelAndJoin()
+        job2.cancelAndJoin()
+
+        coVerify(exactly = 2) { mockLambda1.invoke() }
+        coVerify(exactly = 1) { mockLambda2.invoke() }
     }
 
     @Test
-    fun timerNotStartedUntilStartCalled() = runBlockingTest {
-        var counter = 0
-        val testLambda = { counter += 1 }
+    fun `should stop and restart normally`() = coroutineRule.runBlockingTest {
+        mapboxTimer.routeRefreshInterval = TimeUnit.MINUTES.toMillis(5)
 
-        MapboxTimer(100L, testLambda)
-        advanceTimeBy(200L)
+        // Should emit 1
+        val job1 = mapboxTimer.startRouteRefresh(mockLambda)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(6))
 
-        assertEquals(0, counter)
-    }
+        // Should emit 0 because the jobs are stopped
+        mapboxTimer.stopJobs()
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(30))
 
-    @Test
-    fun multipleStartCalls_doNotExecuteMultipleJobs() = runBlocking {
```

Comment:
This test is not applicable to everything that has a `MapboxTimer`.

Moving this logic so it can be handled by FasterRoute
https://github.com/mapbox/mapbox-navigation-android/pull/2577